src/main/asciidoc/entity-callbacks.adoc

```diff
@@ -0,0 +1,163 @@
+[entity-callbacks]
+= Entity Callbacks
+
+The Spring Data infrastructure provides hooks for modifying an entity before and/or after certain methods are invoked.
+Those so called ``EntityCallback``s provide a convenient way to check and potentially modify an entity in a callback fashioned style. +
+An `EntityCallback` looks pretty much like a specialized `ApplicationListener`, with which you might be already familiar.
```

Comment:
You can leave off ", with which you might be already familiar." Whether they are or not, you want them to read the next sentence, so you don't need to mention familiarity. (If you did keep it, it should be ", with with you might already be familiar" - English gets weird about adverbs).